### ISSUE TITLE
Build failure indexing jars on 1.10.0.CR1

### ISSUE THREAD

**Issue by jorsol** [quarkusio/quarkus#13297]
**Describe the bug**
I have an application that complains about classes not included in the index:
```
[WARNING] [io.quarkus.deployment.steps.ReflectiveHierarchyStep] Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- com.google.common.collect.ImmutableList (sources: JacksonProcessor > com.example.api.commons.config.ResultFormat, JacksonProcessor > com.example.model.Config, JacksonProcessor > com.example.model.Configs, JacksonProcessor > com.example.model.DocParameters, JacksonProcessor > com.example.model.ParametersUpdate, JacksonProcessor > com.example.model.ImmutableParameters$Builder)
	- com.google.common.collect.ImmutableList$Builder (sources: JacksonProcessor > com.example.model.ImmutableConfig$Builder, JacksonProcessor > com.example.model.ImmutableConfigs$Builder, JacksonProcessor > com.example.model.ImmutableParameters$Builder, JacksonProcessor > com.example.model.ImmutableParametersUpdate$Builder)
	- com.google.common.collect.ImmutableMap (sources: JacksonProcessor > com.example.model.History, JacksonProcessor > com.example.model.Units)
	- com.google.common.collect.ImmutableMap$Builder (source: JacksonProcessor > com.example.model.ImmutableUnits$Builder)
	- com.google.common.collect.ImmutableSet (source: JacksonProcessor > com.example.model.ImmutableParameters$Builder)
	- com.google.common.collect.ImmutableSet$Builder (source: JacksonProcessor > com.example.model.ImmutableParameters$Builder)
	- org.antlr.v4.runtime.ParserRuleContext (source: SmallRyeOpenApiProcessor > com.example.exporter.Format)
	- org.antlr.v4.runtime.tree.TerminalNode (source: SmallRyeOpenApiProcessor > com.example.exporter.Format)
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.");.
```
So I include them using `quarkus.index-dependency`, on Quarkus 1.9 this works fine, but on the 1.10.0.CR1 the build fails with the following error:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.10.0.CR1:build (default) on project api: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.IllegalStateException: Class extends type annotation appeared on a non class target
[ERROR] 	at org.jboss.jandex.Indexer.processTypeAnnotation(Indexer.java:538)
[ERROR] 	at org.jboss.jandex.Indexer.processTypeAnnotations(Indexer.java:515)
[ERROR] 	at org.jboss.jandex.Indexer.processAttributes(Indexer.java:355)
[ERROR] 	at org.jboss.jandex.Indexer.processMethodInfo(Indexer.java:306)
[ERROR] 	at org.jboss.jandex.Indexer.index(Indexer.java:1602)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.indexJar(ApplicationArchiveBuildStep.java:308)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.access$200(ApplicationArchiveBuildStep.java:54)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply(ApplicationArchiveBuildStep.java:293)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep$1.apply(ApplicationArchiveBuildStep.java:275)
[ERROR] 	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1224)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.handleJarPath(ApplicationArchiveBuildStep.java:275)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.indexPath(ApplicationArchiveBuildStep.java:191)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.createApplicationArchive(ApplicationArchiveBuildStep.java:185)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.addIndexDependencyPaths(ApplicationArchiveBuildStep.java:172)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.scanForOtherIndexes(ApplicationArchiveBuildStep.java:133)
[ERROR] 	at io.quarkus.deployment.index.ApplicationArchiveBuildStep.build(ApplicationArchiveBuildStep.java:107)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:832)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
``` 

**Configuration**
```properties
quarkus.index-dependency.antlr4.group-id=org.antlr
quarkus.index-dependency.antlr4.artifact-id=antlr4-runtime
quarkus.index-dependency.guava.group-id=com.google.guava
quarkus.index-dependency.guava.artifact-id=guava
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux
 - Output of `java -version`: openjdk version "15.0.1" 2020-10-20
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.10.0.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3

**Comment by samodadela** [comment]
If you change guava 30.0 (quarkus BOM) to guava 30.1-pre, the problem goes away.

**Comment by jorsol** [comment]
It looks like this is fixed with Quarkus 1.11.0.Final
